graphbolt/src/unique_and_compact.h

```diff
@@ -0,0 +1,57 @@
+/**
+ *  Copyright (c) 2023 by Contributors
+ *
+ * @file unique_and_compact.h
+ * @brief Unique and compact op.
+ */
+#ifndef GRAPHBOLT_SHM_UTILS_H_
+#define GRAPHBOLT_SHM_UTILS_H_
+
+#include <torch/torch.h>
+
+namespace graphbolt {
+namespace sampling {
+/**
+ * @brief Removes duplicate elements from the concatenated 'unique_dst_ids' and
+ * 'src_ids' tensor and applies the uniqueness information to compact both
+ * source and destination tensors.
+ *
+ * The function performs two main operations:
+ *   1. Unique Operation: unique(concat(unique_dst_ids, src_ids)), in which the
+ * unique operator will guarantee the unique_dst_ids are at the head of the
+ * result tensor.
+ *   2. Compact Operation: Utilizes the reverse mapping derived from the unique
+ * operation to transform 'src_ids' and 'dst_ids' into condensed IDs, mapping
```

Comment:
We have several terms when dealing with ID mapping, e.g., condensed ID, original ID, inverse ID, and relabeled ID. @frozenbugs Do we have a standard for this kind of names?

Reply:
Let's finalize them and doc it here.
https://docs.google.com/spreadsheets/d/1dv41M-Th20e1ppKvX9wUQ8MFFJKRTiRM6PalYAx9mlk/edit#gid=25977840

Reply:
As the function is called unique_and_compact, use `compacted_ids `for uniform.

Reply:
SG, added to the doc.